karapace/protobuf/syntax_reader.py

```diff
@@ -0,0 +1,367 @@
+        end_pos = min(self.pos + length, len(self.data))
+
+        while self.pos < end_pos:
+            try:
+                digit = int(self.data[self.pos], radix)
+            except ValueError:
+                digit = -1
+
+            if digit == -1 or digit >= radix:
+                break
+
+            if value < 0:
+                value = digit
+            else:
+                value = value * radix + digit
+            self.pos += 1
+
+        self.expect(value >= 0, "expected a digit after \\x or \\X")
+        return chr(value)
+
+    def read_name(self) -> str:
+        """ Reads a (paren-wrapped), [square-wrapped] or naked symbol name. """
+
+        c = self.peek_char()
+        if c == '(':
+            self.pos += 1
+            result = self.read_word()
+            self.expect(self.read_char() == ')', "expected ')'")
+            return result
+        if c == '[':
+            self.pos += 1
+            result = self.read_word()
+            self.expect(self.read_char() == ']', "expected ']'")
+            return result
+        return self.read_word()
+
+    def read_data_type(self) -> str:
+        """ Reads a scalar, map, or type name. """
+
+        name = self.read_word()
+        return self.read_data_type_by_name(name)
+
+    def read_data_type_by_name(self, name: str) -> str:
+        """ Reads a scalar, map, or type name with `name` as a prefix word. """
+        if name == "map":
+            self.expect(self.read_char() == '<', "expected '<'")
+            key_type = self.read_data_type()
+
+            self.expect(self.read_char() == ',', "expected ','")
+            value_type = self.read_data_type()
+
+            self.expect(self.read_char() == '>', "expected '>'")
+            return f"map<{key_type}, {value_type}>"
+        return name
+
+    def read_word(self) -> str:
+        """ Reads a non-empty word and returns it. """
+        self.skip_whitespace(True)
+        start = self.pos
+        while self.pos < len(self.data):
+            c = self.data[self.pos]
+            if ord(c) in range(ord('a'), ord('z') + 1) \
+                    or ord(c) in range(ord('A'), ord('Z') + 1) \
+                    or ord(c) in range(ord('0'), ord('9') + 1) or c in ['_', '-', '.']:
+                self.pos += 1
+            else:
+                break
+        self.expect(start < self.pos, "expected a word")
+        return self.data[start:self.pos]
+
+    def read_int(self) -> int:
+        """ Reads an integer and returns it. """
+        tag: str = self.read_word()
+        try:
+            radix = 10
+            if tag.startswith("0x") or tag.startswith("0X"):
+                radix = 16
+            return int(tag, radix)
+
+        #        except OSError as err:
+        #            print("OS error: {0}".format(err))
+        except ValueError:
+            self.unexpected(f"expected an integer but was {tag}")
+        return -22  # this return never be called but mypy think we need it
```

Comment:
this happens because mypy doesn't know that `unexpected` never returns, the type of that method should be fixed and this line removed

Reply:
ok